File: models/Product.js
const mongoose = require("mongoose");

const productSchema = new mongoose.Schema(
  {
    name: { type: String, required: true },
    price: { type: Number, required: true },
    image: String,
    category: String,
    description: String,

    // 📦 INVENTORY
    stock: {
      type: Number,
      required: true,
      default: 0
    },

    lowStockLimit: {
      type: Number,
      default: 5
    }
  },
  { timestamps: true }   // ✅ ADD THIS
);

module.exports = mongoose.model("Product", productSchema);

File: classy_crafth_app-backup/lib/models/product.dart
class Product {
  final String id;
  final String name;
  final double price;
  final String description;
  final String image;

  Product({
    required this.id,
    required this.name,
    required this.price,
    required this.description,
    required this.image,
  });

  factory Product.fromJson(Map<String, dynamic> json) {
    return Product(
      id: json["_id"] ?? "",
      name: json["name"] ?? "",
      price: double.parse(json["price"].toString()),
      description: json["description"] ?? "",
      image: json["image"] ?? "",
    );
  }
}
